### PR TITLE
Bring the subject shelves back on homepage

### LINKED ISSUE #3552: Bring the subject shelves back on homepage
The subject shelves (see screenshot below) have been removed from the homepage in https://github.com/mozilla/addons-frontend/pull/3460 according to the editorial google doc., but it looks like users like them so we need to add them again (with a slightly different content though).

### PR DESCRIPTION
**Pull request by willdurand**
Fix #3552
Refs #3558

---

This PR adds the subject shelves according to the latest mock-ups (see #3558).

## Screenshots

Before:

screen shot 2017-10-20 at 16 36 03
screen shot 2017-10-20 at 16 36 05

After:

screen shot 2017-10-20 at 16 34 49
screen shot 2017-10-20 at 16 34 51
screen shot 2017-10-20 at 16 37 19
screen shot 2017-10-20 at 16 37 21
screen shot 2017-10-20 at 16 37 26
screen shot 2017-10-20 at 16 37 28

### CODE REVIEW

**Comment by willdurand** [comment]
Note: #3558 will be done in another PR (coming soon). I am aware that the Featured themes shelf is not where it should be.

File: tests/unit/amo/containers/TestHome.js [hunk 1]
```diff
@@ -145,6 +147,27 @@ describe(__filename, () => {
...
+    ];
+
+    const root = render();
+
+    const shelf = root.find('.Home-CuratedThemes');
+    expect(shelf.find('.Home-SubjectShelf-text-wrapper')).toHaveLength(1);
+    expect(shelf.find('.Home-SubjectShelf-list-item'))
+      .toHaveLength(expectedThemes.length);
+    expectedThemes.forEach((slug) => {
+      expect(shelf.find({ to: `/themes/${slug}/` })).toHaveLength(1);
```
Comment (kumar303) [inline tests/unit/amo/containers/TestHome.js]: Cool. I didn't know about property selectors.

**Comment by willdurand** [comment]
> From looking at the Sketch mocks, I think the Themes change how Firefox works... section should come above the featured themes section. I think the explanatory shelves are meant to introduce each type of extension.

@kumar303 thanks! I actually mentioned this in a comment above (not very visible between codecov comment and the PR desc...). I've made various UI changes in #3585.

File: src/amo/components/Home/index.js [hunk 2]
```diff
@@ -68,24 +68,81 @@ export class HomeBase extends React.Component {
...
+        ))}
+      </ul>
+    );
+  }
+
+  renderCuratedThemes() {
+    const { i18n } = this.props;
+    const curatedThemes = [
+      {
+        color: 8,
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/amo/components/Home/index.js]: Yes, I think we'll have to do that for #3594. Our current implementation is leaky.

File: src/amo/components/Home/styles.scss [hunk 3]
```diff
@@ -8,7 +8,54 @@
...
+  .Card-contents {
+    display: flex;
+    flex-flow: row wrap;
+    justify-content: space-between;
+    margin: 0 auto;
+    overflow: auto;
+    width: 100%;
+
+    @include respond-to(medium) {
+      padding: 26px 36px;
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/amo/components/Home/styles.scss]: indeed!

**Comment by willdurand** [comment]
> On a small desktop screen (a bit bigger than tablet) I see scrollbars. Screenshot below.

Good catch! Fixed now :)

**Pull request merged**